### ISSUE TITLE
Detect potential race condition with BrainMonkey test functions

### ISSUE THREAD

**Issue by jrfnl** [Yoast/yoastcs#264]
As discussed with @increddibelly:

1. The potential race condition needs to be reported to BrainMonkey and if possible, fixed there.
2. In the mean time, we want to prevent the Yoast tests running into it.

### Issue outline

> Functions\expect( 'apply_filters' )  should not be used; it causes a race condition with Filters\expectApplied( 'filter_name' ).
> A relatively simple sniff could prevent that.
> The same goes for actions: with expect( 'do_action' ) (bad) and expectDone (good)

### Sniff outline

* Register `T_STRING` and check for content `expect`
* with a `T_NS_SEPARATOR` before it and `T_STRING` `Functions` before that and `T_OPEN_PARENTHESIS` after
* first arg of function call should be `T_CONSTANT_ENCAPSED_STRING` with content `apply_filters` or `do_action`
* check last relevant (function, closure, oo) condition and last condition is a function/closure.
* MAYBE -> is this a function in a test class ? (doesn't need to be a test function, can also be a test helper function) -> _eigenlijk niet relevant -> check bij schrijven sniff of het een verschil maakt_
* if so, check if `expectApplied` of `expectDone` is called within the same function
    in other words: `T_STRING` met content `expect...` + followed by open parenthesis

### Planning

=> YoastCS 2.3.0 + tag @increddibelly as reviewer + inform Irene about release

**Comment by increddibelly** [comment]
@herregroen clarified that it's not Filters\expect( 'apply_filters' ) but Functions\expect. It looks like BrainMonkey also mocks `apply_filters` itself when you call `expectApplied` but when you also mock `apply_filters` by yourself only one of these mocks will succeed.
